### PR TITLE
fix: remove intermidiate step of login while registering

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes https://github.com/fossasia/open-event-frontend/issues/5433

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/controllers/register.js [hunk 1]
```diff
@@ -22,11 +22,8 @@ export default class RegisterController extends Controller {
     this.model.save()
       .then(user => {
         this.set('session.newUser', user.get('email'));
-        if (this.inviteToken) {
-          this.send('loginExistingUser', user.get('email'), password, this.inviteToken, this.event);
-        } else {
-          this.transitionToRoute('login');
-        }
+        this.send('loginExistingUser', user.get('email'), password, this.inviteToken, this.event);
+        this.transitionToRoute('/');
```
Comment (iamareebjamal) [inline app/controllers/register.js:26]: Why is this needed? Shouldn't the above action be awaited?

File: app/controllers/register.js [hunk 2]
```diff
@@ -60,7 +57,9 @@ export default class RegisterController extends Controller {
             await this.store.findRecord('user', tokenPayload.identity)
           );
         }
-        this.transitionToRoute('public.role-invites', eventId, { queryParams: { token } });
+        if (token != null) {
+          this.transitionToRoute('public.role-invites', eventId, { queryParams: { token } });
+        }
```
Comment (iamareebjamal) [inline app/controllers/register.js:62]: Shouldn't the `this.transitionToRoute('/');` call be placed here instead in else condition

File: app/controllers/register.js (hunk 1 above)
Comment (maze-runnar) [reply app/controllers/register.js:26]: > Shouldn't the above action be awaited?

why? I am not getting it.

File: app/controllers/register.js (hunk 2 above)
Comment (maze-runnar) [reply app/controllers/register.js:63]: ok

File: app/controllers/register.js [hunk 3]
```diff
@@ -60,7 +56,11 @@ export default class RegisterController extends Controller {
             await this.store.findRecord('user', tokenPayload.identity)
           );
         }
-        this.transitionToRoute('public.role-invites', eventId, { queryParams: { token } });
+        if(token != null) {
```
Comment (iamareebjamal) [inline app/controllers/register.js:59]:
```suggestion
        if (token) {
```
